ipatests/test_ipalib/test_filestore.py

```diff
@@ -0,0 +1,985 @@
+#
+# Copyright (C) 2018  FreeIPA Contributors see COPYING for license
+#
+
+"""
+Module provides unit tests to verify that the FileStore code works.
+"""
+
+from __future__ import absolute_import
+from hashlib import sha256
+from ipaplatform.paths import paths
+
+try:
+    # install subpackages are not shipped with wheels
+    from ipalib.install import sysrestore
+    from ipalib.install.sysrestore import SYSRESTORE_INDEXFILE as INDEX
+    from ipalib.install.sysrestore import SYSRESTORE_MAX_INDEX as MAX_INDEX
+    from ipalib.install.sysrestore import SYSRESTORE_PATH_INDEX as PATH_INDEX
+    from ipalib.install.sysrestore import SYSRESTORE_SEP as SEP
+except ImportError:
+    INDEX = MAX_INDEX = PATH_INDEX = SEP = None
+
+import collections
+import filecmp
+import os
+import pytest
+import shutil
+import six
+import uuid
+
+# pylint: disable=import-error
+if six.PY3:
+    # The SafeConfigParser class has been renamed to ConfigParser in Py3
+    from configparser import ConfigParser as SafeConfigParser
+    from configparser import MissingSectionHeaderError
+else:
+    from ConfigParser import SafeConfigParser
+    from ConfigParser import MissingSectionHeaderError
+# pylint: enable=import-error
+
+# install subpackages are not shipped with wheels
+pytestmark = [pytest.mark.needs_install_package, pytest.mark.tier0]
+
+STORE_ROOT_DIR = os.path.join(paths.TMP, "sysrestore")
```

Comment:
The handling of temporary files here, in setup_module, and setup_function are problematic. Please use the global fixture ``tempdir``, which will give you a unique temporary directory that will be removed afterwards.

Reply:
Fixed.